### DISCUSSION TITLE
/blog/2022/06/23/long-labels-ggplot/

### DISCUSSION THREAD

**Discussion by andrewheiss** [andrewheiss/ath-quarto#50] [Blog comments]
Quick and easy ways to deal with long labels in ggplot2 | Andrew Heiss 
 https://www.andrewheiss.com/blog/2022/06/23/long-labels-ggplot/

**Comment by andrewheiss** [comment]
```header
Originally posted by andreashandel on 2022-09-12 13:33:36
````
Hi Andrew, I just happened to need this and was trying to implement, but
got a bit stuck. In my case, I need special expressions (e.g. Greek
letters) on the labels. I can\'t figure out to combine that with the
label_wrap() command. Do you happen to know how to do that? Here is a
small reprex. In the example below, I\'d like to line-break x and y-axes
and the variable labels. So far, haven\'t had much luck, figured maybe
you know:

\
`## ---- packages --------`\
`library(ggplot2)`\
`library(dplyr)`\
\
`## ---- createdata --------`\
`N = 100 #sample size`\
`# made-up data`\
`df &lt;- data.frame(BMI = runif(N,15,50),`\
`STATUS = as.factor(rbinom(N,3,prob=0.4)),`\
`Y = rnorm(N,100,10) )`\
\
`# giving the STATUS variable some silly labels`\
`status_labels &lt;- c(expression(paste(&quot;Healthy male, receiving &quot;, Omega[2], &quot; treatment&quot;) ),`\
`expression(paste(&quot;Immunocompromised male, not receiving &quot;, Omega[2], &quot; treatment&quot;)),`\
`expression(paste(&quot;Healthy female: No ❤ problems&quot;)),`\
`expression(paste(&quot;Everyone else who doesn't fit into the other categories.&quot; )) )`\
\
`## ---- plot0 --------`\
`p0 &lt;- ggplot(df) + geom_boxplot(aes(x = STATUS, y = Y)) +`\
`geom_point(aes(x = STATUS, y = Y,color = BMI)) +`\
`labs(x = expression(paste(&quot;Patient status according to &quot;, Omega[2], &quot; treatment, ❤-condition and other factors&quot;))) +`\
`labs(y = expression(paste(&quot;Response after &quot;, 21^2,&quot; days excluding those days on which it didn't rain.&quot;))) +`\
`scale_x_discrete(labels = parse(text=status_labels))`\
`plot(p0)`\

**Reply by andrewheiss** [reply]
```header
Originally posted on 2022-09-12 17:49:41
````
It seems that \`label_wrap()\` and \`str_wrap()\` aren\'t compatible
with \`expression()\`. However, you can fake it if you use ggtext since
it can parse a subset of HTML tags like \&lt;sub\&gt; and \&lt;sup\&gt;!

Here\'s a reprex:
[https://gist.github.com/and\...](https://gist.github.com/andrewheiss/a01092503b01438bb0a2d82af3aec738)

[https://uploads.disquscdn.c\...](https://uploads.disquscdn.com/images/b716a06d6a2a5aa5a214c3ffd810a6a5e628eb847d289550fd3477386d85f356.png)

**Reply by andrewheiss** [reply]
```header
Originally posted by andreashandel on 2022-09-12 18:22:59
````
Thanks! Not quite as elegant as label_wrap() but it works (at least for
my use case)! Feel free to add this to your post or make a new one :)

**Comment by IrshadUlHaq1** [comment]
This is a great way to handle this very common challenge. I wonder if there is a way to do the same for facet_wrap labels without increasing the width of figure?

**Reply by andrewheiss** [reply]
Yep! You can use `ggplot2::label_wrap_gen()` for that without needing anything from {scales}:

```r
library(tidyverse)

df <- tibble(
  x = rnorm(100),
  y = rnorm(100),
  facet = rep(paste0("Here's a really long label ", 1:2), each = 50)
)

ggplot(df, aes(x = x, y = y)) +
  geom_point() +
  facet_wrap(vars(facet), labeller = labeller(facet = label_wrap_gen(20)))
```

![image](https://github.com/andrewheiss/ath-quarto/assets/73663/2a9f1ff2-a5ac-4e21-90af-665789388536)

**Comment by ngreifer** [comment]
Came across this nice post today and wanted to add a modification to the rotated labels version that I think can look quite nice, which is to have the end of each label align with the axis tic by setting `hjust = 0`, as follows:

``` r
ggplot(essential_by_category,
       aes(x = CATEGORY, y = total)) +
  geom_col() +
  scale_y_continuous(labels = comma) +
  labs(x = NULL, y = "Total projects") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

![](https://i.imgur.com/axiqw0B.png)

Loved this post and I learned something new! Just wanted to add another option I have found useful.
